tutor/src/screens/teacher-gradebook/table.js

```diff
@@ -282,22 +282,29 @@ const StudentColumnHeader = observer(({ ux }) => {
   );
 });
 
-const AssignmentHeading = observer(({ ux, heading, sortKey }) => {
-  const onClick = () => ux.changeRowSortingOrder(sortKey, 'score');
-  return (
-    // Overlay has a lot of problems when showing at the top. Putting at the bottom for now
+const AssignmentHeading = observer(({ ux, heading }) => {
+  // get the ref of the heading-title to see if it is wrapped with ellipsis
+  // if so, show the tooltip, otherwise don't show it.
+  const titleTextRef = useRef();
+  const [showToolTip, setShowToolTip] = useState(true);
+  useEffect(() => {
+    setShowToolTip(titleTextRef.current.offsetWidth < titleTextRef.current.scrollWidth);
+  });
+
+  return (   
     <OverlayTrigger
+      // Overlay has a lot of problems when showing at the top. Putting at the bottom for now
       placement="bottom"
```

Comment:
I usually prefer to omit placement so it's `auto` - that way it'll adjust the position so it's shown regardless of the page scroll.

Reply:
Yes, but I think the first choice is the `top` placement which causes a bug. That is why I set it to `bottom`. The bug seems to be caused by some of the cells using the `column` css property which caused a lot of bugs already in other places.